components/backup-stream/src/subscription_manager.rs

```diff
@@ -0,0 +1,445 @@
+                }),
+            )
+            .map_err(|err| {
+                annotate!(
+                    err,
+                    "failed to send request to region info accessor, server maybe too too too busy. (region id = {})",
+                    region.get_id()
+                )
+            })?;
+        let new_region_info = rx
+            .recv()
+            .map_err(|err| annotate!(err, "BUG?: unexpected channel message dropped."))?;
+        if new_region_info.is_none() {
+            metrics::SKIP_RETRY
+                .with_label_values(&["region-absent"])
+                .inc();
+            return Ok(());
+        }
+        let new_region_info = new_region_info.unwrap();
+        if new_region_info.role != StateRole::Leader {
+            metrics::SKIP_RETRY.with_label_values(&["not-leader"]).inc();
+            return Ok(());
+        }
+        let removed = self.subs.deregister_region(&region, |old, _| {
+            let should_remove = old.handle().id == handle.id;
+            if !should_remove {
+                warn!("stale retry command"; "region" => ?region, "handle" => ?handle, "old_handle" => ?old.handle());
+            }
+            should_remove
+        });
+        if !removed {
+            metrics::SKIP_RETRY
+                .with_label_values(&["stale-command"])
+                .inc();
+            return Ok(());
+        }
+        metrics::INITIAL_SCAN_REASON
+            .with_label_values(&["retry"])
+            .inc();
+        self.start_observe(region).await;
+        Ok(())
+    }
+
+    async fn get_last_checkpoint_of(&self, task: &str) -> Result<TimeStamp> {
+        let meta_cli = self.get_meta_client();
+        let last_checkpoint = TimeStamp::new(meta_cli.global_progress_of_task(task).await?);
+        Ok(last_checkpoint)
+    }
+
+    fn observe_over_with_initial_data_from_checkpoint(
+        &self,
+        region: &Region,
+        last_checkpoint: TimeStamp,
+        handle: ObserveHandle,
+    ) {
+        self.subs
+            .register_region(region, handle.clone(), Some(last_checkpoint));
+
+        // we should not spawn initial scanning tasks to the tokio blocking pool
+        // beacuse it is also used for converting sync File I/O to async. (for now!)
+        // In that condition, if we blocking for some resouces(for example, the `MemoryQuota`)
+        // at the block threads, we may meet some ghosty deadlock.
+        let s = self.scan_pool_handle.send(ScanCmd {
+            region: region.clone(),
+            handle,
+            last_checkpoint,
+        });
+        if let Err(err) = s {
+            annotate!(err, "BUG: scan_pool closed").report(format!(
+                "during initial scanning for region {}",
+                region.get_id()
+            ));
+        }
+    }
+
+    fn find_task_by_region(&self, r: &Region) -> Option<String> {
+        self.range_router
+            .find_task_by_range(&r.start_key, &r.end_key)
```

Comment:
will a region include multiple task in the future?

Reply:
At that time, the rule of this should be changed.

Reply:
please Leave a NOTE or TODO here. to tell others we do not support multiple tasks right now, one day we implement multiple tasks we should changed it.